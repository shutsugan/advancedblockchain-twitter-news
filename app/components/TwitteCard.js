import React, {Component} from 'react';
import {
    AppRegistry, 
    Text, 
    View, 
    TouchableHighlight, 
    StyleSheet,
    Dimensions,
    Button, 
    Linking 
} from 'react-native';
import {Card, SocialIcon} from 'react-native-elements';
import {Constants} from 'expo';

export default class TwitteCard extends Component {
    render() {
        const {hash, twitter, date, title, body, link} = this.props.twitte;
        
        return(
            <TouchableHighlight style={styles.twittecard__container}>
                <View style={styles.twittecard}>
                    <Card
                        overlayStyle={{ opacity: 0 }}
                        containerStyle={{ width: Dimensions.get('window').width }}
                    >
                        <Text style={styles.twittecard__date}>{date}</Text>
                        <Text></Text>
                        <Text style={styles.twittecard__title}>{title}</Text>
                        <Text style={styles.twittecard__body}>{body}</Text>

                        <Text 
                            style={styles.twittecard__link} 
                            onPress={_ => Linking.openURL(link)}>{link}
                        </Text>

                        <SocialIcon
                            title={`Follow ${hash}`}
                            button
                            type='twitter'
                            onPress={_ => Linking.openURL(twitter)}
                        />
                    </Card>
                </View>
            </TouchableHighlight>
        );
    }
}

const width = Dimensions.get('window').width;

const styles = StyleSheet.create({
    twittecard__container: {
        width: width,
        backgroundColor: '#e5ded8',
    },

    twittecard: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        paddingTop: Constants.statusBarHeight,
        marginBottom: 16,
    },

    twittecard__title: {
        color: '#000',
        fontSize: 18,
    },

    twittecard__date: {
        color: '#9197A3',
        fontSize: 11,
        fontWeight: 'bold',
    },

    twittecard__body: {
        color: '#9197A3',
        fontSize: 14,
    },

    twittecard__link: {
        alignSelf: 'flex-start',
        fontSize: 12,
        marginTop: 8,
        marginBottom: 16,
        color: '#54acef',
        borderBottomColor: '#54acef',
        borderBottomWidth: 1,
    }
});

AppRegistry.registerComponent('TwitteCard', _ => TwitteCard);