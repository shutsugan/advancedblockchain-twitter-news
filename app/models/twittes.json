[
	{
		"title": "Advanced Blockchain AG",
		"hash": "@BlockchainAG",
		"twitter": "https://twitter.com/BlockchainAG",
		"date": "15. 02. 2018",
		"body": "Who invented #Bitcoin ? There's a lot more to the question than you might think.",
		"link": "http://bit.ly/2GxAUSV"
	},
	{
		"title": "Advanced Blockchain AG",
		"hash": "@BlockchainAG",
		"twitter": "https://twitter.com/BlockchainAG",
		"date": "11. 02. 2018",
		"body": "Just as terms changed and took on new meaning as the internet evolved, we may see similar progressions in the crypto world. Either way, staying informed is key. Do you know the difference between #blockchain and #DLT?",
		"link": "http://bit.ly/2ETIxpv"
	},
	{
		"title": "Advanced Blockchain AG",
		"hash": "@BlockchainAG",
		"twitter": "https://twitter.com/BlockchainAG",
		"date": "08. 02. 2018",
		"body": "Der Aktionär TV met up with @social_nakamoto founder and Advanced Blockchain AG CVO, Robert A. Küfner, to discuss the evolution of DLT and the @iotatoken Project. @deraktionaertv #Blockchain #IOTA #Tangle Here's the full interview:",
		"link": "http://bit.ly/2ETIxpv"
	},
	{
		"title": "Advanced Blockchain AG",
		"hash": "@BlockchainAG",
		"twitter": "https://twitter.com/BlockchainAG",
		"date": "27. 11. 2017",
		"body": "The first blog in the @social_nakamoto Knowledge Base series sheds light on the evolution of the ledger, and the natural next step - #DLT. #Blockchain #Tangle 'What is DLT?' is available here:",
		"link": "https://bitly.com/"
	},
	{
		"title": "Advanced Blockchain AG",
		"hash": "@BlockchainAG",
		"twitter": "https://twitter.com/BlockchainAG",
		"date": "27. 11. 2017",
		"body": "The #Blockchain, #Tangle & Beyond event series took the Advanced Blockchain AG team to North America. CEO Michael Geike gave his first presentations of 2018 alongside #IOTA Developers and members of the IOTA Foundation. @iotatoken",
		"link": "http://bit.ly/2r4p8MT"
	}
]