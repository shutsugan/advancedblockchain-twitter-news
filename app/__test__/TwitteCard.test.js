import React from 'react';
import renderer from 'react-test-renderer';

import TwitteCard from '../components/TwitteCard';
import twittes from '../models/twittes.json';

function getRandomInt(max) {
  return Math.floor(Math.random() * Math.floor(max));
}

function getRandomTwitte(twittes, random_int) {
	return twittes[random_int];
}

const twitte = getRandomTwitte(twittes, 0);

it('TwitteCard renders without crashing', () => {
  const rendered = renderer.create(<TwitteCard twitte={twitte} />).toJSON();
  expect(rendered).toBeTruthy();
});

it('TwitteCard renders correctly', () => {
  const tree = renderer.create(<TwitteCard twitte={twitte} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('TwitteCard should be a View', () => {
  const tree = renderer.create(<TwitteCard twitte={twitte} />).toJSON();
  const twittesList_type = tree.type;

  expect(twittesList_type).toEqual('View');
});

it('TwitteCard has children', () => {
  const tree = renderer.create(<TwitteCard twitte={twitte} />).toJSON();
  const children = tree.children.length > 0 ? true : false;

  expect(children).toBeTruthy();
});

it('TwitteCard has one child', () => {
  const tree = renderer.create(<TwitteCard twitte={twitte} />).toJSON();
  const children_length = tree.children.length;

  expect(children_length).toEqual(1);
});

it('TwitteCard child should be a View', () => {
  const tree = renderer.create(<TwitteCard twitte={twitte} />).toJSON();
  const child = tree.children.shift().type;

  expect(child).toEqual('View');
});

it('TwitteCard twitte should be contained in twittes', () => {
  const tree = <TwitteCard twitte={twitte} />;
  const props_twitte = tree.props.twitte;

  expect(twittes).toContain(props_twitte)
});