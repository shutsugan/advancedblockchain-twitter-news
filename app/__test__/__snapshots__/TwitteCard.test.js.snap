// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwitteCard renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e5ded8",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 16,
        "paddingTop": 20,
      }
    }
  >
    <View
      overlayStyle={
        Object {
          "opacity": 0,
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#e1e8ee",
            "borderWidth": 1,
            "margin": 15,
            "marginBottom": 0,
            "padding": 15,
            "shadowColor": "rgba(0,0,0, .2)",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          Object {
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#9197A3",
              "fontSize": 11,
              "fontWeight": "bold",
            }
          }
        >
          15. 02. 2018
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "fontSize": 18,
            }
          }
        >
          Advanced Blockchain AG
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#9197A3",
              "fontSize": 14,
            }
          }
        >
          Who invented #Bitcoin ? There's a lot more to the question than you might think.
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "flex-start",
              "borderBottomColor": "#54acef",
              "borderBottomWidth": 1,
              "color": "#54acef",
              "fontSize": 12,
              "marginBottom": 16,
              "marginTop": 8,
            }
          }
        >
          http://bit.ly/2GxAUSV
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "shadowColor": "rgba(0,0,0, .4)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 1,
                "shadowRadius": 1,
              },
              Object {
                "alignItems": "center",
                "borderRadius": 30,
                "flexDirection": "row",
                "justifyContent": "center",
                "margin": 7,
              },
              Object {
                "paddingBottom": 14,
                "paddingTop": 14,
              },
              false,
              false,
              Object {
                "backgroundColor": "#00aced",
              },
              undefined,
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 24,
                  },
                  Array [
                    undefined,
                  ],
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              ï‚™
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontWeight": "bold",
                      "marginLeft": 15,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Follow @BlockchainAG
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
